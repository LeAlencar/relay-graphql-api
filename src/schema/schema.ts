import { GraphQLSchema } from 'graphql';

import QueryType from './QueryType';
import MutationType from './MutationType';
import TransactionNewSubscription from './SubscriptionType';

export const schema = new GraphQLSchema({
  query: QueryType,
  mutation: MutationType,
  subscription: TransactionNewSubscription
});
