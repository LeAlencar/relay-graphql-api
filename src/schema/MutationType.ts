import { GraphQLObjectType } from 'graphql';

import TransactionMutations from '../modules/transaction/mutations';
import UserMutations from '../modules/user/mutations'

export default new GraphQLObjectType({
  name: 'Mutation',
  description: "Root of ... mutations",
  fields: () => ({
    ...TransactionMutations,
    ...UserMutations
  }),
});
