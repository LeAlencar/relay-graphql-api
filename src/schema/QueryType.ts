
import { GraphQLObjectType } from 'graphql';
import { connectionArgs, connectionFromArray } from 'graphql-relay';
import { TransactionConnection } from '../modules/transaction/TransactionType';
import { nodeField } from '../modules/node/NodeInterface'
import TransactionModel from '../modules/transaction/TransactionModel';

const QueryType = new GraphQLObjectType({
    name: 'Query',
    description: 'The root of all queries',
    fields: () => ({
        node: nodeField,
        transactions: {
            type: TransactionConnection,
            args: {
              ...connectionArgs
            },
            resolve: async (_, args) => {
                const data = await TransactionModel.find({})
                return connectionFromArray(data, args)
            }
        }
    })
})
export default QueryType