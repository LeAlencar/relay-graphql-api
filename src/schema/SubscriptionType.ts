import { GraphQLObjectType } from 'graphql';
import TransactionNewSubscription from '../modules/transaction/subscriptions/newTransactionSubscription';



const SubscriptionType = new GraphQLObjectType({
  name: 'Subscription',
  fields: {
    TransactionNew: TransactionNewSubscription as any ,
  },
});

export default SubscriptionType;
