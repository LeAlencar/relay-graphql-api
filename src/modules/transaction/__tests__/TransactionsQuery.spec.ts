import { graphql } from 'graphql';

import { clearDbAndRestartCounters, connectMongoose, disconnectMongoose, sanitizeTestObject } from '../../../../test';
import { createUser } from '../../user/fixture/createUser';
import { schema } from '../../../schema/schema';
import { getContext } from '../../../getContext';
import { createTransaction } from '../fixture/createTransaction';

beforeAll(connectMongoose);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

it('should return transaction from logged user only', async () => {
  const user = await createUser();

  await createTransaction({
    owner: user,
  });


  // language=GraphQL
  const query = `
    query Q {
      transactions(first: 10) {
        edges {
          node {
            name
            owner {
              id
              username
            }
            price
            category
          }
        }
      }
    }
  `;

  const rootValue = {};
  const contextValue = await getContext({ user });
  const variableValues = {};

  const result = await graphql({
    schema,
    source: query,
    rootValue,
    contextValue,
    variableValues,
  });

  expect(result.errors).toBeUndefined();

  // eslint-disable-next-line
  console.log('result: ', result.data.transactions.edges[0].node);
  expect(result.data.transactions.edges.length).toBe(1);
  expect(sanitizeTestObject(result.data)).toMatchSnapshot();
});
