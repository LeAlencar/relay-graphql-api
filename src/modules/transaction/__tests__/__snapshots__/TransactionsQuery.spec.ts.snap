// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return transaction from logged user only 1`] = `
{
  "transactions": {
    "edges": [
      {
        "node": {
          "category": "Category 0",
          "name": "Transaction#0",
          "owner": {
            "id": "FROZEN-ID",
            "username": "user#1",
          },
          "price": "100.00",
        },
      },
    ],
  },
}
`;
